control-base/src/main/java/fi/nls/oskari/control/data/LayerAndGroupOrderHandler.java

```diff
@@ -0,0 +1,172 @@
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.map.MaplayerGroup;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+/**
+ * CRUD for layer and group order handling. Methods require admin user.
+ */
+@OskariActionRoute("LayerAndGroupOrder")
+public class LayerAndGroupOrderHandler extends RestActionHandler {
+	private static Logger log = LogFactory.getLogger(LayerAndGroupOrderHandler.class);
+	
+    private static final String KEY_PARENT = "parent";
+    private static final String KEY_ORDERS = "orders";
+    private static final String KEY_TYPE = "type";
+    private static final String KEY_ID = "id";
+    private static final String KEY_OLD_PARENT = "oldParent";
+    private static final String TYPE_LAYER = "layer";
+	
+	private OskariMapLayerGroupService oskariMapLayerGroupService;
+	private OskariLayerService oskariLayerService;
+	
+	/**
+	 * Variable to keep track of the new order of the groups.
+	 */
+	private HashMap<Integer, Integer> newGroupOrderMap = new HashMap<>();
+	/**
+	 * Variable to keep track of the new order of the layers.
+	 */
+	private HashMap<Integer, Integer> newLayerOrderMap = new HashMap<>(); 
+
+    public void setOskariMapLayerGroupService(final OskariMapLayerGroupService service) {
+        oskariMapLayerGroupService = service;
+    }
+    
+    public void setOskariLayerService(final OskariLayerService service) {
+    	oskariLayerService = service;
+    }
+
+    @Override
+    public void init() {
+        // setup service if it hasn't been initialized
+        if(oskariMapLayerGroupService == null) {
+            setOskariMapLayerGroupService(new OskariMapLayerGroupServiceIbatisImpl());
+        }
+        // setup service if it hasn't been initialized
+        if(oskariLayerService == null) {
+        	setOskariLayerService(new OskariLayerServiceIbatisImpl());
+        }
+        if(newGroupOrderMap == null) {
+        	newGroupOrderMap = new HashMap<>();
+        }
+        if(newLayerOrderMap == null) {
+        	newLayerOrderMap = new HashMap<>();
+        }
+    }
+
+    /**
+     * Handles updating the order and group of the given node i.e. layer or group.
+     * @param params
+     * @throws ActionException
+     */
+    @Override
+    public void handlePost(ActionParameters params) throws ActionException {
+        checkForAdminPermission(params);
+        log.debug("Updating layer/group order");
+        JSONObject orderJSON = getOrderJSON(params.getRequest());
+
+
+		try {
+			int parentID = orderJSON.getInt(KEY_PARENT);
+			JSONArray orders = orderJSON.getJSONArray(KEY_ORDERS);
+
+			// change main groups orders
+			if(parentID == -1){
```

Comment:
This is a bit difficult to read. Could we separate the blocks in this if-else to separate functions? Like 
```
if (parentID == -1) {
    updateGroupOrder(orders);
} else {
    updateLayerGroups(orders);
}
```